core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/OAbstractPaginatedStorage.java

```diff
@@ -3291,7 +3291,14 @@ public Object command(final OCommandRequestText iCommand) {
           executor.setProgressListener(iCommand.getProgressListener());
           executor.parse(iCommand);
 
-          return executeCommand(iCommand, executor);
+          String currentThreadName = Thread.currentThread().getName();
+          try {
+            Thread.currentThread().setName(currentThreadName + " <command>" + iCommand + "</command>");
+            return executeCommand(iCommand, executor);
+          } finally {
+            Thread.currentThread().setName(String.valueOf(currentThreadName));
```

Comment:
Why String.valueOf, the parameter is already String, is not it?

Reply:
is null value accepted as a thread name...? In case we can just pass the value as it is